modin/conftest.py

```diff
@@ -567,76 +573,73 @@ def s3_base(worker_id):
-            # in the popen command and we throw an error within the body of the context
-            # manager, the test just hangs forever.
-            with subprocess.Popen(
-                # try this https://stackoverflow.com/a/72084867/17554722 ?
-                shlex.split(f"moto_server s3 -p {endpoint_port}"),
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
-            ) as proc:
-                made_connection = False
-                for _ in range(50):
-                    try:
-                        # OK to go once server is accepting connections
-                        if requests.get(endpoint_uri).ok:
-                            made_connection = True
-                            break
-                    except Exception:
-                        # try again while we still have retries
-                        time.sleep(0.1)
-                if not made_connection:
-                    raise RuntimeError(
-                        "Could not connect to moto server after 50 tries."
-                    )
-                yield endpoint_uri
-
-                proc.terminate()
+            # assume CI has started moto in docker container:
+            # https://docs.getmoto.org/en/latest/docs/server_mode.html#run-using-docker
+            # It would be nice to start moto on another thread as in the
+            # instructions here:
+            # https://docs.getmoto.org/en/latest/docs/server_mode.html#start-within-python
+            # but that gives 403 forbidden error when we try to create the bucket
+            yield "http://localhost:5000"
+    else:
+        # Launching moto in server mode, i.e., as a separate process
+        # with an S3 endpoint on localhost
+
+        # If we hit this else-case, this test is being run locally. In that case, we want
+        # each worker to point to a different port for its mock S3 service. The easiest way
+        # to do that is to use the `worker_id`, which is unique, to determine what port to point
+        # to. We arbitrarily assign `5` as a worker id to the master worker, since we need a number
+        # for each worker, and we never run tests with more than `pytest -n 4`.
+        worker_id = "5" if worker_id == "master" else worker_id.lstrip("gw")
+        endpoint_port = f"555{worker_id}"
+        endpoint_uri = f"http://127.0.0.1:{endpoint_port}/"
+
+        # pipe to null to avoid logging in terminal
+        # TODO any way to throw the error from here? e.g. i had an annoying problem
+        # where I didn't have flask-cors and moto just failed .if there's an error
+        # in the popen command and we throw an error within the body of the context
+        # manager, the test just hangs forever.
+        with subprocess.Popen(
+            # try this https://stackoverflow.com/a/72084867/17554722 ?
+            shlex.split(f"moto_server s3 -p {endpoint_port}"),
```

Comment:
is there a reason _not_ to use an explicit list?
```suggestion
            ["moto_server", "s3", "-p", str(endpoint_port)],
```

Reply:
This is how it is implemented in pandas, at first glance it seems superfluous to use `shlex`. Let's try without it.